django/contrib/gis/gdal/raster/source.py

```diff
@@ -425,6 +425,21 @@ def warp(self, ds_input, resampling='NearestNeighbour', max_error=0.0):
 
         return target
 
+    def clone(self):
+        "Return a clone of this GDALRaster object"
+        return GDALRaster(
+            capi.copy_ds(
+                self.driver._ptr,
+                force_bytes(self.name),
```

Comment:
Clone shouldn't share a file with source, see [GDAL docs](https://gdal.org/tutorials/raster_api_tut.html#using-createcopy):

_"A simple copy from the a file named pszSrcFilename, to a new file named pszDstFilename ..."_

Reply:
I added fix and test.